### PR TITLE
Adds 'IN' metadata filter for pgvector for checking set presence

### PR DESCRIPTION
**Pull request by jadespanning**
# Adds "IN" metadata filter for pgvector to all checking for set presence

PGVector currently supports metadata filters of the form:
```
{"filter": {"key": "value"}}
```
which will return documents where the "key" metadata field is equal to "value".

This PR adds support for metadata filters of the form:
```
{"filter": {"key": { "IN" : ["list", "of", "values"]}}}
```

Other vector stores support this via an "$in" syntax. I chose to use "IN" to match postgres' syntax, though happy to switch.
Tested locally with PGVector and ChatVectorDBChain.

@dev2049

### CODE REVIEW

**Comment by dev2049** [comment]
nice! could we add a simple integration test?

**Comment by jadespanning** [comment]
Added an integration test!

File: langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -291,8 +291,15 @@ def similarity_search_with_score_by_vector(
         if filter is not None:
             filter_clauses = []
             for key, value in filter.items():
-                filter_by_metadata = EmbeddingStore.cmetadata[key].astext == str(value)
-                filter_clauses.append(filter_by_metadata)
+                IN = "IN"
```
Comment (eyurtsev) [inline langchain/vectorstores/pgvector.py:294]: could we use a lower case `in` -- I think it's responsible to use `snake_case` given that we're working in python.

@hwchase17 / @agola11 is there a common representation for filters?

File: langchain/vectorstores/pgvector.py [hunk 2]
```diff
@@ -291,8 +291,15 @@ def similarity_search_with_score_by_vector(
...
+                if isinstance(value, str):
+                    filter_by_metadata = EmbeddingStore.cmetadata[
+                        key].astext == str(value)
+                    filter_clauses.append(filter_by_metadata)
+                elif isinstance(value, dict) and IN in value:
+                    filter_by_metadata = EmbeddingStore.cmetadata[
+                        key].astext.in_(value[IN])
+                    filter_clauses.append(filter_by_metadata)
```
Comment (eyurtsev) [inline langchain/vectorstores/pgvector.py:302]: I think there's a bug in the code w/ respect to handling of `int` / `float` values. 

In the previous version: 

{ 'page': 5 }

Would've resulted in a filter being issued with the number `5` being converted to the string literal "5".

In the new version, no filter clause will be generated.

To fix the behavior, could you put the `dict` check as the first `if statement`, and use the previous behavior for the `else` branch that will use the str conversion logic for everything that's not a dict?

File: langchain/vectorstores/pgvector.py (hunk 1 above)
Comment (jadespanning) [reply langchain/vectorstores/pgvector.py]: switched to lower case, and made the matching case insensitive.

mainly made this choice to match with postgres, which allows any case.

File: langchain/vectorstores/pgvector.py (hunk 2 above)
Comment (jadespanning) [reply langchain/vectorstores/pgvector.py:303]: done

**Comment by eyurtsev** [comment]
ready for merge, maintain needs to resolve linting issues

**Comment by dev2049** [comment]
landed #4982 !

**Comment by devilankur18** [comment]
@jadespanning please add this to docs

**Pull request closed without merging**